port_ocean/core/event_listener/once.py

```diff
@@ -41,6 +45,34 @@ def __init__(
     ):
         super().__init__(events)
         self.event_listener_config = event_listener_config
+        self.resync_state: dict[str, Any] = {}
+
+    def should_update_resync_state(self) -> bool:
+        return ocean.config.runtime == RuntimeType.Saas.value
+
+    async def before_resync(self) -> None:
+        if not self.should_update_resync_state():
+            return None
+
+        now = datetime.datetime.now()
+        try:
+            integration = await ocean.port_client.get_current_integration()
+            interval_str = (
+                integration.get("spec", {})
+                .get("appSpec", {})
+                .get("scheduledResyncInterval")
+            )
+            interval = convert_time_to_minutes(interval_str)
+            self.resync_state["next_resync"] = calculate_next_resync(now, interval)
+        except Exception:
+            logger.exception("Error occurred while calculating next resync")
+            return None
+
+    async def after_resync(self) -> None:
+        if not self.should_update_resync_state():
+            return None
+
+        await ocean.port_client.update_resync_state(self.resync_state)
```

Comment:
I want this implemented a bit different 
With a lifecycle context manager

Reply:
Seems like a great idea!
but I think it's out of this PR's scope and it will inflate this PR more than it should, WDYT?